bindings/c/test/mako/async.cpp

```diff
@@ -125,15 +125,18 @@ void ResumableStateForRunWorkload::runOneTick() {
 				postStepFn(f, tx, args, key1, key2, val);
 			if (iter.stepKind() != StepKind::ON_ERROR) {
 				if (auto err = f.error()) {
-					logr.printWithLogLevel(err.retryable() ? VERBOSE_WARN : VERBOSE_NONE,
+					logr.printWithLogLevel(err.retryable() || (args.transaction_timeout > 0 && err.is(1031 /*timeout*/))
```

Comment:
There are a lot places with repeating "(err.retryable() || (args.transaction_timeout > 0 && err.is(1031 /*timeout*/)"

Let's maybe make it more readable by moving it to:
ResumableStateForRunWorkload::isExpectedError(const Error& err)

Reply:
This would make sense. But I think if we do something like this, we should also change it for the sync mode for which I would probably try to move the logging logic out of future.hpp. Would it make sense to completely remove the logging logic from future.hpp and let the calling appications handle the logging? I think this would be cleaner. Then we also wouldn't need the `timeouts_expected` parameter.

Reply:
In the sync mode the functions are global, so we would need some more refactoring to do an analogous design. I think such refactoring is beyond the scope of this PR.